src/libcore/iter.rs

```diff
@@ -2423,7 +2425,11 @@ macro_rules! step_impl {
             #[allow(trivial_numeric_casts)]
             fn steps_between(start: &$t, end: &$t, by: &$t) -> Option<usize> {
                 if *start <= *end {
-                    Some(((*end - *start) / *by) as usize)
+                    // Note: We assume $t <= usize/isize here
+                    Some(
+                        ((*end as isize).wrapping_sub(*start as isize) as usize
```

Comment:
Ugh, my patch is totally broken because I've been thinking *only* about the signed case. Need to account for the unsigned case too, I understand if it's puzzling to read.